test/Microsoft.AspNetCore.SystemWebAdapters.Tests/HttpServerUtilityTests.cs

```diff
@@ -85,6 +86,8 @@ public void UrlTokenRoundtrip(string input, string expected)
     [InlineData("/api/test/request/info", "/UploadedFiles", "UploadedFiles")]
     [InlineData("/api/test/request/info", "UploadedFiles", "api", "test", "request", "UploadedFiles")]
     [InlineData("/api/test/request/info", "~/MyUploadedFiles", "MyUploadedFiles")]
+    [InlineData("/api/test/request/info", "~/TrailingSlash/", "TrailingSlash\\")]
```

Comment:
```suggestion
    [InlineData("/api/test/request/info", "~/", "\\")]
    [InlineData("/api/test/request/info", "~/TrailingSlash/", "TrailingSlash\\")]
```

should we also be testing this?